pkg/generate/app/cmd/newapp.go

```diff
@@ -585,9 +645,66 @@ func makeImageStreamKey(ref kapi.ObjectReference) string {
 	return types.NamespacedName{ref.Namespace, name}.String()
 }
 
+// Search executes the provided config and returns the result of the resolution.
+func (c *AppConfig) RunSearch(out io.Writer) (*SearchResult, error) {
+	c.ensureDockerSearcher()
+	repositories, err := c.individualSourceRepositories()
+	if err != nil {
+		return nil, err
+	}
+	err = c.detectSource(repositories)
+	if err != nil {
+		return nil, err
+	}
+	components, repositories, environment, parameters, err := c.validate()
+	if err != nil {
+		return nil, err
+	}
+
+	switch {
+	case len(repositories) == 0 && len(components) == 0:
+		return nil, ErrNoInputs
+	case len(repositories) > 0:
```

Comment:
I think you need to split this switch into separate ifs. Each case has to be checked separately because they can be combined in a new-app call.